ddp-workspace/projects/ddp-dsm-ui/src/app/scanner/scanner.component.ts

```diff
@@ -67,24 +69,24 @@ export class ScannerComponent implements DoCheck, OnDestroy {
 
   public removeFields(index: number): void {
     this.scannerFields.length - 1 && this.scannerFields.removeAt(index);
+    this.shortIds.length - 1 && this.shortIds.splice(index, 1);
     this.activeScanner.inputFields.forEach((inputField: InputField) =>
       this.checkForDuplicates(inputField.controllerName));
   }
 
   public save(): void {
     this.additionalMessage = '';
     const filteredActiveScannerFieldsGroupsArray = this.filteredNonNullFieldsGroups;
-
+    this.shortIds = Array(...Array(this.scannerFields.length)).map(function() {});
     this.activeScanner.saveFn(filteredActiveScannerFieldsGroupsArray)
       .pipe(
         takeUntil(this.subscriptionSubject$)
       )
       .subscribe({
         next: (data: any[]) => {
           this.cdr.markForCheck();
-          if (data.some(d => d)) {
+          if (data.some(d => d?.hasOwnProperty('error') || d?.hasOwnProperty('shortId'))) {
```

Comment:
When the response object includes the `shortId` field and we don't remove those scan pairs, maybe better to create a separate function that would only be responsible for setting a success message under those scan pairs, as I think it would be more convenient, because we don't remove scan pairs, hence we don't need to perform that action under the function called removeSuccessfulScans.

**For example:**
To remove that function and do something like this:
```
if (data.some(d => d?.hasOwnProperty('error') || d?.hasOwnProperty('shortId'))) {
            this.checkResponse(data); // could be better name
          } else {
            this.resetForm();
            this.additionalMessage = 'Data saved';
          }

...

public checkResponse(responseData: any[]): void {
responseData.forEach((data: any, index: number) => {
    if(data?.hasOwnProperty('error')) {
        this.setError(index) // would set error to the formGroup at passed index
    }  else if(data?.hasOwnProperty('shortId')) {
         this.shortIdSuccess(index) // would set or store successfully messages, which later would be rendered in the UI
       }
      }
    );

  this.removeSuccessfullySavedFields() // would remove other successful pairs, as per legacy logic
}
```

What do you think?